pallets/asset/src/lib.rs

```diff
@@ -1520,6 +1646,11 @@ impl<T: Trait> Module<T> {
 
         Self::_update_checkpoint(ticker, from_portfolio.did, from_total_balance);
         Self::_update_checkpoint(ticker, to_portfolio.did, to_total_balance);
+        if let Some(timestamp) = Self::is_checkpoint_due(ticker) {
```

Comment:
I think this needs to be moved up a few lines (before the `_update_checkpoint` calls above it).

Reply:
`is_checkpoint_due` only uses the part of state which is not accessed by `_update_checkpoint` above. So `is_checkpoint_due` could be moved above but that wouldn't make difference to the result.